File: 백준/Bronze/2991. 사나운 개/사나운 개.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int a, b, c, d, cnt;
    vector<int> arr(3);
    cin >> a >> b >> c >> d;
    for (int i = 0; i < 3; i++) {
        cin >> arr[i];
        cnt = 0;
        if (0 < arr[i] % (a + b) && arr[i] % (a + b) <= a)
            cnt += 1;
        if (0 < arr[i] % (c + d) && arr[i] % (c + d) <= c)
            cnt += 1;
        cout << cnt << "\n";
    }
    return 0;
}

File: 백준/Bronze/2991. 사나운 개/사나운 개.py
import sys
input=lambda:sys.stdin.readline().rstrip()
a,b,c,d=map(int,input().split())
arr=[*map(int,input().split())]
for n in arr:
    cnt=0
    if 0<n%(a+b)<=a:
        cnt+=1
    if 0<n%(c+d)<=c:
        cnt+=1
    print(cnt)